### DISCUSSION TITLE
What is the key binding for running Rust code?

### DISCUSSION THREAD

**Discussion by anupj** [LunarVim/LunarVim#3010] [General]
I can't figure out the keyboard keymap for :RustRun, in the keymap setting it says <D-R>. What does that mean ? which keys should I press on the keyboard to execute :RustRun? Thanks.

![Screenshot 2022-09-11 at 05 14 04](https://user-images.githubusercontent.com/177374/189512140-fba45c80-bf0c-450f-ac21-d48179941aa2.png)

Bonus: Where can I find all the bindings for rust lang?

**Comment by pdamianik** [comment]
![image](https://user-images.githubusercontent.com/39028343/202564727-4098666d-d2bb-4fce-aea5-f0156adf1f41.png)

These are the keyboard shortcuts i got from my stock lunarvim config. After a quick look into the neovim help i found that `<D-...>` should map to `command` under mac and `super`/`windows` key under other platforms.

![image](https://user-images.githubusercontent.com/39028343/202566723-63192f95-1484-4971-bf0c-faebf45de900.png)

But hitting `super+r` doesn't do anything on my setup (I am using linux) so I looked a bit further into it and found the following issue and commit:

https://github.com/neovim/neovim/issues/2190
https://github.com/justinmk/neovim/commit/99d4c8c29c4a9371c268cc20e4805709d86fb686

The commit mentions the following:

> For non-GUI (terminal) support, user must:
>  `:set <D-a>={CSI sequence}`
> then send the {CSI sequence} from their terminal. But this does not work
> yet (regression https://github.com/neovim/neovim/issues/2204).

I also tried running the keyboard shortcut with [neovide/neovide](https://github.com/neovide/neovide) but even then it didn't work.
For now it seems that non-mac users can only remap `<D-...>` binds (mentioned by https://github.com/neovim/neovim/issues/2190#issuecomment-160421715) using https://www.lunarvim.org/docs/configuration/keybindings

**Reply by anupj** [reply]
Thanks for the detailed reply 🙏🏼
